src/filesystem_stream.h

```diff
@@ -0,0 +1,170 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef EP_FILESYSTEM_STREAM_H
+#define EP_FILESYSTEM_STREAM_H
+
+// Headers
+#include <cassert>
+#include <istream>
+#include <ostream>
+#include "utils.h"
+
+namespace Filesystem_Stream {
+	class InputStream : public std::istream {
```

Comment:
Since these stream objects are going to be used direct and not behind a pointer, should we make them `final` so people don't try to inherit from them and get slicing behavior?

Also if this is final. the `size` member can be `private` and not `protected`

Reply:
The ``protected`` was a logic error. I thought the Filesystem-Implementations will inherit from it but they will all only implement there own buffer objects, so this can be made ``final+private``